packages/svelte/messages/compile-warnings/style.md

```diff
@@ -1,3 +1,7 @@
 ## css_unused_selector
 
 > Unused CSS selector "%name%"
+
+Svelte traverses both the template and the `<style>` tag to find out which of the CSS selectors are actually in use. It will prune (i.e. remove) those selectors which appear unused and warn you about it.
+
+In some situations Svelte may be too strict and you _know_ that the selector is used. In this case, you can use [`:global` or `:global(...)`](global-styles) to prevent them from being pruned.
```

Comment:
this makes it sound like Svelte is buggy, rather than this being a deliberate design

````suggestion
Svelte traverses both the template and the `<style>` tag to find out which of the CSS selectors are not used within the template, so it can remove them.

In some situations a selector may target an element that is not 'visible' to the compiler, for example because it is part of an `{@html ...}` tag or you're overriding styles in a child component. In these cases, use [`:global`](/docs/svelte/global-styles) to preserve the selector as-is:

```svelte
<div class="post">{@html content}</div>

<style>
  .post :global {
    p {...}
  }
</style>
```
````
```suggestion
Svelte traverses both the template and the `<style>` tag to find out which of the CSS selectors are actually in use. It will prune (i.e. remove) those selectors which appear unused and warn you about it.

In some situations Svelte may be too strict and you _know_ that the selector is used. In this case, you can use [`:global` or `:global(...)`](global-styles) to prevent them from being pruned.
```